src/rules/ResourceQuoteStyle.js

```diff
@@ -253,18 +259,20 @@ class ResourceQuoteStyle extends ResourceRule {
         const targetQuotesAllAlt = this.localeOnly ?
             targetQuotesNativeAlt :
             new RegExp(`((^|\\W)[${targetQuoteStartAlt}']\\s?[\\p{Letter}\\{]|[\\p{Letter}\\}]\\s?[${targetQuoteEndAlt}'](\\W|$))`, "gu");
-    
-        // the non quote chars are used to highlight errors in the target string
+
+        // the non quote chars are used to highlight errors in the target string where they are using quotes, but
+        // they are the wrong type. Start with the superset of all quotes and then remove the valid ones so that
+        // you are left with the wrong ones for this locale.
         const targetNonQuoteChars = quoteChars.
                 replace(sourceQuoteStart, "").
                 replace(targetQuoteStart, "").
                 replace(sourceQuoteEnd, "").
                 replace(targetQuoteEnd, "");
         const targetNonQuoteCharsAlt = quoteChars.
-                replace(sourceQuoteStartAlt, "").
-                replace(targetQuoteStartAlt, "").
-                replace(sourceQuoteEndAlt, "").
-                replace(targetQuoteEndAlt, "");
+                replace(new RegExp(`[${sourceQuoteStartAlt}]`, "gu"), "").
+                replace(new RegExp(`[${targetQuoteStartAlt}]`, "gu"), "").
+                replace(new RegExp(`[${sourceQuoteEndAlt}]`, "gu"), "").
+                replace(new RegExp(`[${targetQuoteEndAlt}]`, "gu"), "");
```

Comment:
why do we need to instantiate a regexp here?

Reply:
replace() will replace a whole string with another string. When that string is a single character, that is fine, but in this case, we have a regular expression that we need to replace with the "global" flag turned on.